src/lib.rs

```diff
@@ -452,6 +597,12 @@ impl<D, T> Message<D, T> {
             .duration_since(UNIX_EPOCH)
             .expect("Time went backwards");
 
+        let topic = match (hedwig.message_router)(&data_type, &data_schema_version.0) {
+            Some(t) => t,
+            None => return Err(MessageError::RouterError("Topic not found")),
```

Comment:
you can you use `ok_or(Err(MessageError::RouterError("Topic not found"))` here instead of the match. name is a bit misleading because it is an option not a result but 🤷‍♂

Reply:
ah, that is cleaner, thanks!